matrix_webcam/__main__.py

```diff
@@ -85,91 +97,112 @@ def main() -> None:
     # background is a matrix of the actual letters (not lit up) -- the underlay.
     # foreground is a binary matrix representing the position of lit letters -- the overlay.
     # dispense is where new 'streams' of lit letters appear from.
-    background = rand_string(printable.strip(), width * height)
+    printable_chars = printable.strip()
+    background = rand_string(printable_chars, width * height)
     foreground: list[tuple[int, int]] = []
     dispense: list[int] = []
 
     delta = 0
     bg_refresh_counter = random.randint(3, 7)
     perf_counter = time.perf_counter()
+    frame_counter = 0
+    time_counter = time.monotonic_ns()
+    fps = 0
+
+
+    #pre-allocate some arrays
+    # Source image array
+    _, image = cap.read()
+
+    # Segmentation resolution arrays
+    imageseg = np.empty((144, 256, 3), dtype=np.uint8)
+    condition = np.empty((144, 256), dtype=bool)
+
+    # Final resolution arrays
+    imager = np.empty((height, width, 3), dtype=np.uint8)
+    gray = np.empty((height, width), dtype=np.uint8)
+    interp = interp_method((144, 256), (height, width))
 
-    bg_image: Optional[npt.NDArray[np.uint8]] = None
     with SelfieSegmentation(model_selection=1) as selfie_segmentation:
         while cap.isOpened():
-            success, image = cap.read()
+            success, image = cap.read(image)
 
             if not success:
                 print("Ignoring empty camera frame.")
-                # If loading a video, use 'break' instead of 'continue'.
                 continue
+            image = cv2.flip(image, 1, image)
```

Comment:
left hand assignment is not need btw

Reply:
I realize that, however, my stylistic choice is to have that assignment as I think it makes it easier to catch mistakes.